### PR TITLE
Created HLEDatabase for XactEng 5849

### PR DESCRIPTION
**Pull request by LukeUsher**

### CODE REVIEW

**Comment by LukeUsher** [comment]
This addresses a tiny part of #705

**Comment by darrena092** [comment]
Looks great. Just checked the OOVPA against XACTEngineCreate at $30462F in Kingdom Under Fire The Crusaders and it fits! Will test later on.

**Comment by jarupxx** [comment]
Good job. XACTEngineCreate and XACTEngineDoWork hits in Shikigami no Shiro II.

**Comment by RadWolfie** [comment]
If you guys don't mind my asking... did the titles made further progress?

@LukeUsher I can confirm there is no false detection for XACTEngineCreate and XACTEngineDoWork OOVPA signature do match. However, XACTEngine is using DirectSound directly. Which makes this not really necessary? Unless others do confirm it has made further progress. Then we can keep filling in for 5849 database of XACTEngine signatures if needed.

File: src/CxbxKrnl/HLEDataBase/XactEng.1.0.5849.inl [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+OOVPA_NO_XREF(XACTEngineCreate, 5849, 11)
+		// XACTEngineCreate+0x09 : movzx ebx, al
+		{ 0x09, 0x0F },
+		{ 0x0A, 0xB6 },
+		{ 0x0B, 0xD8 },
+
+        // XACTEngineCreate+0x2B : push edi
+		{ 0x2B, 0x57 },
+
+		// XACTEngineCreate+0x31 : push 0x168
```
Comment (RadWolfie) [inline src/CxbxKrnl/HLEDataBase/XactEng.1.0.5849.inl]: Just a note to others: This is very unique offset/value which made it impossible to make false detection within XAct header section. Great job Luke! 👍

**Comment by LukeUsher** [comment]
Portions of xact access audio hardware registers directly, without going through DirectSound, so it does need to be patched.

**Comment by RadWolfie** [comment]
I see... interesting finding Luke, is that all you plan on to merge into master? Or are you planning on add more into 5849 database?

**Comment by LukeUsher** [comment]
I'm planning a merge, the LLE work I'm doing at the moment is taking up an insane amount of time, I just figured I'd get this started to get the ball rolling

**Pull request merged**